src/DayPickerInput.js

```diff
@@ -562,18 +562,33 @@ export default class DayPickerInput extends React.Component {
     const { inputProps } = this.props;
     return (
       <div className={this.props.classNames.container}>
-        <Input
-          ref={el => (this.input = el)}
-          placeholder={this.props.placeholder}
-          {...inputProps}
-          value={this.state.typedValue || this.state.value}
-          onChange={this.handleInputChange}
-          onFocus={this.handleInputFocus}
-          onBlur={this.handleInputBlur}
-          onKeyDown={this.handleInputKeyDown}
-          onKeyUp={this.handleInputKeyUp}
-          onClick={!inputProps.disabled ? this.handleInputClick : undefined}
-        />
+        {this.props.component &&
+        {}.toString.call(this.props.component) !== '[object Function]' ? (
```

Comment:
Why not just `typeof this.props.component === 'function'`?

Reply:
This way is more verbose.  Not really relevant here but one good example is that it will return `Array` for `Array` instead of `Object`.  A better way would be to actually have a function that would extract out `Function`.